packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -619,7 +616,7 @@ export async function activate(extensionContext: vscode.ExtensionContext) {
     telemetryService,
     services: {
       ChannelService,
-      TelemetryService,
+      TelemetryProvider,
```

Comment:
I think this is a fine approach.  We'll need to discuss with the E4D team and see if they want the clean break here in the API, or if we have to be backwards compatible.